File: Python/Easy/الأعداد الأولية بين قيمتين.py

# Creator: OmarAfet
# https://profile.satr.codes/OmarAfet/public/overview
# https://github.com/OmarAfet

import math
def getPrimesBetween(a: int, b: int) -> list[int]:
    if a < 2:
        a = 2
    if b < a:
        return []

    primes = []

    for num in range(a, b+1):
        is_prime = True

        for i in range(2, int(math.sqrt(num))+1):
            if num % i == 0:
                is_prime = False
                break

        if is_prime:
            primes.append(num)

    return primes

File: JavaScript/Easy/الأعداد الأولية بين قيمتين.js
/*

Creator: OmarAfet
https://profile.satr.codes/OmarAfet/public/overview
https://github.com/OmarAfet

*/

function getPrimesBetween(a, b) {
	if (a < 2) {
		a = 2;
	}
	if (b < a) {
		return [];
	}

	let primes = [];

	for (let num = a; num <= b; num++) {
		let is_prime = true;

		for (let i = 2; i <= Math.sqrt(num); i++) {
			if (num % i == 0) {
				is_prime = false;
				break;
			}
		}

		if (is_prime) {
			primes.push(num);
		}
	}

	return primes;
}
